src/src/com/microsoft/aad/adal/StorageHelper.java

```diff
@@ -466,7 +466,7 @@ final synchronized private SecretKey getSecretKeyFromAndroidKeyStore() throws IO
             final byte[] encryptedKey = readKeyData(keyFile);
             sSecretKeyFromAndroidKeyStore = unwrap(wrapCipher, encryptedKey);
             Logger.v(TAG, "Finished reading SecretKey");
-        } catch (Exception ex) {
+        } catch (GeneralSecurityException ex) {
             // Reset KeyPair info so that new request will generate correct KeyPairs.
             // All tokens with previous SecretKey are not possible to decrypt.
             Logger.e(TAG, "Unwrap failed for AndroidKeyStore", "", ADALError.ANDROIDKEYSTORE_FAILED);
```

Comment:
@imakhalova I saw your email mentioning there are many decryption failure from android keystore, and i saw the unwrap failure you mentioned. We don't even have the exception passed into the logger, then there is no way we know the exception stacktrace. You don't need to worry about it in the PR, I'm going through the whole encryption/decryption logic. I will make the change and send another PR.